pennylane/interfaces/jax.py

```diff
@@ -15,24 +15,47 @@
 This module contains functions for adding the JAX interface
 to a PennyLane Device class.
 """
-# pylint: disable=too-many-arguments
-import inspect
+# pylint: disable=unused-argument
 import logging
+from typing import Tuple, Callable
+
+import dataclasses
 
 import jax
 import jax.numpy as jnp
 
 import pennylane as qml
 from pennylane.transforms import convert_to_numpy_parameters
-
-from .jacobian_products import _compute_jvps
+from pennylane.typing import ResultBatch
 
 dtype = jnp.float64
 
 logger = logging.getLogger(__name__)
 logger.addHandler(logging.NullHandler())
 
 
+Batch = Tuple[qml.tape.QuantumTape]
+ExecuteFn = Callable[[Batch], qml.typing.ResultBatch]
+
+
+@dataclasses.dataclass
+class NonPytreeWrapper:
```

Comment:
Do we need this visible in the docs? If this is something that only internal PennyLane developers are expected to see then please could we not have it in the docs.

Reply:
did a documentation pass on the draft.